apps/studio/components/ui/CopyButton.tsx

```diff
@@ -30,8 +44,13 @@ const CopyButton = ({
   return (
     <Button
       onClick={async (e) => {
+        let textToCopy = text
```

Comment:
preference for immutability

`const textToCopy = asyncText ? await asyncText() : text`

Reply:
also we'll probably need a loader if its async?

Reply:
it's so fast it would create a flash.
perhaps add one if the async takes longer than a specified time.

do in a follow up tho